### PR TITLE
Replicate domain

### PR DESCRIPTION
**Pull request by wxing1292**
solve #569

### CODE REVIEW

File: service/frontend/handler.go [hunk 1]
```diff
@@ -273,15 +300,69 @@ func (wh *WorkflowHandler) UpdateDomain(ctx context.Context,
...
+	configVersion := getResponse.ConfigVersion
 	failoverVersion := getResponse.FailoverVersion
+	currentCluster := clusterMetadata.GetCurrentClusterName()
 
 	// whether active cluster is changed
 	activeClusterChanged := false
 	// whether anything other than active cluster is changed
 	configurationChanged := false
+
+	validateReplicationConfig := func(existingDomain *persistence.GetDomainResponse,
```
Comment (samarabbas) [inline service/frontend/handler.go]: nit: Why this validation is defined as a closure? Can we just have a validation function defined on the handler like we do for all different kind of validation?

File: service/frontend/handler.go [hunk 2]
```diff
@@ -273,15 +300,69 @@ func (wh *WorkflowHandler) UpdateDomain(ctx context.Context,
...
+			}
+			replicationConfig.ActiveClusterName = currentCluster
+		}
+
+		// validate active cluster is also specified in all clusters
+		activeClusterInClusters := false
+		for _, cluster := range replicationConfig.Clusters {
+			if cluster.ClusterName == replicationConfig.ActiveClusterName {
+				activeClusterInClusters = true
+				break
```
Comment (samarabbas) [inline service/frontend/handler.go]: one coding practice we follow is to always use label when breaking out of loops.

File: schema/cadence/versioned/v0.6/manifest.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
```
[the remark this answers is not in the document]
Comment (wxing1292) [reply schema/cadence/versioned/v0.6/manifest.json]: we have not release that yet, however, 0.5 is rolled out to staging.
we can definitely do some manual work on staging.
let me know your preference

File: service/frontend/handler.go (hunk 1 above)
Comment (wxing1292) [reply service/frontend/handler.go]: the reason i put this validation as a closure: this function is not reusable, by other logics.

but sure, i will make the change

File: service/frontend/handler.go (hunk 2 above)
Comment (wxing1292) [reply service/frontend/handler.go]: ok sure

File: schema/cadence/versioned/v0.6/manifest.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (wxing1292) [reply schema/cadence/versioned/v0.6/manifest.json]: fixed, changed to 0.5

**Pull request merged**